packages/golden-layout/src/container/ItemContainer.ts

```diff
@@ -146,14 +151,32 @@ export default class ItemContainer<
    * Closes the container if it is closable. Can be called by
    * both the component within at as well as the contentItem containing
    * it. Emits a close event before the container itself is closed.
+   * @param options Options to pass into the beforeClose handler
    */
-  close() {
-    if (this._config.isClosable) {
+  close(options?: CloseOptions) {
+    if (this.beforeCloseHandler) {
+      // Only close if the beforeCloseHandler returns true
+      if (this.beforeCloseHandler(options)) {
+        if (this._config.isClosable) {
+          this.emit('close');
+          this.parent.close();
+        }
+      }
+    } else if (this._config.isClosable) {
+      // If there is no handler, proceed as normal and close if the container is closable
       this.emit('close');
       this.parent.close();
     }
   }
 
+  /**
+   * Sets the beforeCloseHandler to callback
+   * @param callback Callback function to call before closing
+   */
+  beforeClose(callback: (options?: CloseOptions) => boolean) {
```

Comment:
Need to allow `null` to be passed in here as well, so before handler can be removed if necessary (by passing `beforeClose(null)`), e.g.:
```
beforeClose(callback: ((options?: CloseOptions) => boolean) | null) {
```